### PR TITLE
Woodcutting: Enchanted Lumberfall

### PR DESCRIPTION
**Pull request by 8SIXSect**
## Describe your changes
- Everytime a player uses **Tree Feller**, they will now have a chance to get special items (like how _Fishing_ does it but without legendaries).
- What happens is one of the leaves gets removed, a small particle is spawned, and the random special item (like a rune, diamond block, stick, etc.) is placed in the player's inventory.
- But this perk also allows the player to double their drops like how _Feeling Lucky_ works
- Note: I gave the skill 250 levels but the woodcutting perk menu is really cluttered and I'll fix it eventually

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested my changes.

### CODE REVIEW

File: progression/src/main/java/me/mykindos/betterpvp/progression/profession/woodcutting/WoodcuttingLoadConfig.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.bukkit.Material
+import org.bukkit.configuration.ConfigurationSection
+import java.util.*
+
+
+/**
+ * - Loads the YAML configuration for the Woodcutting Profession & returns the loaded data
+ * - This function will cause side effects by logging messages to the console
+ */
+fun loadConfigForWoodcutting(config: ExtendedYamlConfiguration,
```
Comment (Mykindos) [inline progression/src/main/java/me/mykindos/betterpvp/progression/profession/woodcutting/WoodcuttingLoadConfig.kt:15]: Any particular reason you did this in kotlin? we have plenty of examples in java of loading config this way

**Review by Mykindos — CHANGES_REQUESTED** [review]

**Comment by ezrn** [comment]
Should it put it in the players inventory? or should it drop the special item

File: progression/src/main/java/me/mykindos/betterpvp/progression/profession/woodcutting/WoodcuttingLoadConfig.kt (hunk 1 above)
Comment (8SIXSect) [reply progression/src/main/java/me/mykindos/betterpvp/progression/profession/woodcutting/WoodcuttingLoadConfig.kt:15]: I thought the code would be more concise if put in Kotlin but I can change it to Java if want me to

**Comment by 8SIXSect** [comment]
> Should it put it in the players inventory? or should it drop the special item

Fishing drops the special item but fishing also doesn't have to account for obstructions being in the way. When you fish, it's in an open area with few blocks preventing you from getting the item. Even when dropping it close to the trunk of the tree, trying to find the special item felt tedious because the terrain and leaves get in the way

**Comment by braulio-dev** [comment]
Actual chad for using Kotlin, you have my stamp of approval

File: progression/src/main/java/me/mykindos/betterpvp/progression/profession/woodcutting/WoodcuttingLoadConfig.kt (hunk 1 above)
Comment (Mykindos) [reply progression/src/main/java/me/mykindos/betterpvp/progression/profession/woodcutting/WoodcuttingLoadConfig.kt:15]: To keep it consistent with the other 99.9% of the project, its probably best to keep it in java

**Review by Mykindos — APPROVED** [review]

**Pull request merged**